src/components/document/editable-document-content.tsx

```diff
@@ -6,7 +6,8 @@ import { CanvasComponent } from "./canvas";
 import { DocumentContextReact } from "./document-context";
 import { FourUpComponent } from "../four-up";
 import { useDocumentContext } from "../../hooks/use-document-context";
-import { useGroupsStore, useUIStore, useUserStore } from "../../hooks/use-stores";
+import { useDocumentSyncToFirebase } from "../../hooks/use-document-sync-to-firebase";
+import { useGroupsStore, useStores, useUIStore, useUserStore } from "../../hooks/use-stores";
```

Comment:
A few unused imports here.

Reply:
Fixed in an upcoming commit. Also, I lied about not having any more changes 😬 , but I think having reviewed the current code is worthwhile. Leslie asked about supporting automatic retry of saves and I realized that I could refactor the current approach one level further and make use of ReactQuery's `useMutation` hook to accomplish the same thing as before but in a more reusable package. So expect one more commit shortly.